src/sys/windows/selector.rs

```diff
@@ -623,29 +624,60 @@ cfg_net! {
         }
     }
 
-    fn get_base_socket(raw_socket: RawSocket) -> io::Result<RawSocket> {
+    fn try_get_base_socket(raw_socket: RawSocket, ioctl: u32) -> Result<RawSocket, i32> {
         let mut base_socket: RawSocket = 0;
         let mut bytes: u32 = 0;
-
         unsafe {
             if WSAIoctl(
                 raw_socket as usize,
-                SIO_BASE_HANDLE,
+                ioctl,
                 null_mut(),
                 0,
                 &mut base_socket as *mut _ as PVOID,
                 size_of::<RawSocket>() as u32,
                 &mut bytes,
                 null_mut(),
                 None,
-            ) == SOCKET_ERROR
+            ) != SOCKET_ERROR
             {
-                Err(io::Error::last_os_error())
-            } else {
                 Ok(base_socket)
+            } else {
+                Err(WSAGetLastError())
             }
         }
     }
+
+    fn get_base_socket(raw_socket: RawSocket) -> io::Result<RawSocket> {
+        let res = try_get_base_socket(raw_socket, mswsock::SIO_BASE_HANDLE);
+        if let Ok(base_socket) = res {
+            return Ok(base_socket);
+        }
+
+        // The `SIO_BASE_HANDLE` should not be intercepted by LSPs, therefore
```

Comment:
Nit: I don't know what "LSP" is in this context, but if its a well known term you can ignore this comment.

Reply:
Layered Service Provider
https://en.wikipedia.org/wiki/Layered_Service_Provider
Good luck, you still don't know what it is 😉